runtime/metricsview/executor_rewrite_time.go

```diff
@@ -41,6 +73,47 @@ func (e *Executor) resolveTimeRange(ctx context.Context, tr *TimeRange, tz *time
 		return nil
 	}
 
+	if tr.RillTime == "" {
+		return e.resolveTimeRangeLegacy(ctx, tr, tz, executionTime)
+	}
+
+	rt, err := rilltime.Parse(tr.RillTime)
+	if err != nil {
+		return err
+	}
+
+	t, err := e.loadWatermark(ctx, executionTime)
+	if err != nil {
+		return err
+	}
+
+	minTime, err := e.GetMinTime(ctx, e.metricsView.TimeDimension)
+	if err != nil {
+		return err
+	}
+
+	tr.Start, tr.End, err = rt.Resolve(rilltime.ResolverContext{
+		Now:        time.Now(),
+		MinTime:    minTime,
+		MaxTime:    t,
```

Comment:
The watermark is not necessarily `MaxTime` – it may be, but users can override it to something else (e.g. to avoid partial data). 

The design doc says to differentiate between the `earliest`, `latest`, `now` and `watermark` constants: https://www.notion.so/rilldata/Time-range-and-time-range-series-syntax-0be92cca367a425682bb5ea09032b260?pvs=4#131ba33c8f578010b792cac30273e818

Reply:
Huh. I thought according Mike we only used `earliest` and `latest`. Lemme add watermark as a separate thing.